### PR TITLE
fix: Zip export download link not working

### LINKED ISSUE #5555: Event Export: Zip export download link not working
After generating a zip file the download on the website is not working. The user receives an automated email with a download link, which is working. However, on the website itself there is an empty error message box and nothing happens.

Reproduce:
1. Create an event with details
2. Go to "Export" and export an event
3. Click Download button.

Screenshot from 2020-11-10 10-32-40

### PR DESCRIPTION
**Pull request by sachinchauhan2889**
Fixes #5555 

#### Checklist

- [X] I have read the [Contribution & Best practices Guide](https://blog.fossasia.org/open-source-developer-guide-and-best-practices-at-fossasia).
- [X] My branch is up-to-date with the Upstream `development` branch.
- [X] The acceptance, integration, unit tests and linter pass locally with my changes 
- [X] I have added tests that prove my fix is effective or that my feature works
- [X] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: app/components/events/view/export/download-zip.js [hunk 1]
```diff
@@ -5,25 +5,30 @@ import { action } from '@ember/object';
...
-      const anchor = document.createElement('a');
-      anchor.style.display = 'none';
-      anchor.href = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
-      anchor.download = 'EventExport.zip';
-      anchor.click();
-      this.notify.success(this.l10n.t('Exported Event Downloaded successfully.'), {
-        id: 'export_succ'
+    this.loader
+      .downloadFile(`${eventDownloadUrl}`)
+      .then(res => {
```
Comment (iamareebjamal) [inline app/components/events/view/export/download-zip.js]: Please use async await, revert this

File: app/components/events/view/export/download-zip.js [hunk 2]
```diff
@@ -5,25 +5,23 @@ import { action } from '@ember/object';
...
-      anchor.style.display = 'none';
-      anchor.href = URL.createObjectURL(new Blob([res], { type: 'octet/stream' }));
-      anchor.download = 'EventExport.zip';
-      anchor.click();
-      this.notify.success(this.l10n.t('Exported Event Downloaded successfully.'), {
-        id: 'export_succ'
+    this.loader
+      .downloadFile(`${eventDownloadUrl}`)
+      .then(() => {
+        this.notify.success(this.l10n.t('Exported Event Downloaded successfully.'), {
```
Comment (iamareebjamal) [inline app/components/events/view/export/download-zip.js:13]: Please use async await. Revert this

File: app/components/events/view/export/download-zip.js (hunk 1 above)
Comment (sachinchauhan2889) [reply app/components/events/view/export/download-zip.js]: ok i will.

File: app/services/loader.js [hunk 3]
```diff
@@ -37,12 +37,13 @@ export default Service.extend({
     data = data || config.data || null;
 
     if (!config.isExternal) {
-      if (!url.startsWith('/')) {
-        url = `/${url}`;
+      if (!url.startsWith('http')) {
+        if (!url.startsWith('/')) {
+          url = `/${url}`;
+        }
+        url = config.withoutPrefix ? `${adapter.host}${url}` : `${adapter.urlPrefix()}${url}`;
```
Comment (iamareebjamal) [inline app/services/loader.js:44]: What is it doing?

Comment (sachinchauhan2889) [reply app/services/loader.js:44]: actually sir problem was here. 
export explain

while download anything like invioce, ticket etc, server send dowload url of this type ----> /v1/orders/123456ased/download.pdf.. loader take this download url and append it after api address and make it kike this -----> https://api.eventyay.con/v1/orders/123456ased/dowload.pdf...

but when we comw to download zip, server give full address like----> https://api.evetyay.com/v1/export/custom/........, and when we use loader.downloadFile it apeend it after api address and make like this--->   https://api.eventyay.com/https://api.eventyay.com/v1/export..... this is why download is nnot working.

to elove this i make a check before appending downloadurl after host api. 
if downloadurl starts with hhtp then no need to append downloadurl after host api, leave it as it is.

that's why i done this. 
sir, if you have better approach suggest me.

Comment (iamareebjamal) [reply app/services/loader.js:44]: Simple solution is to use `isExternal: true`

Comment (sachinchauhan2889) [reply app/services/loader.js:44]: Ok sir, i got it. I will do this

**Comment by sachinchauhan2889** [comment]
@iamareebjamal sir, ready for review

**Review by iamareebjamal — DISMISSED** [review]

File: app/components/events/view/export/download-zip.js [hunk 4]
```diff
@@ -8,7 +8,7 @@ export default class DownloadZip extends Component {
   async exportEventDownload(eventDownloadUrl) {
     this.set('isLoading', true);
     try {
-      const res = await this.loader.downloadFile(`${eventDownloadUrl}`);
+      const res = await this.loader.downloadFile(`${eventDownloadUrl}`, null, { isExternal: true });
```
Comment (iamareebjamal) [inline app/components/events/view/export/download-zip.js:11]:
```suggestion
      const res = await this.loader.downloadFile(eventDownloadUrl, null, { isExternal: true });
```

**Pull request merged**